faq.rst

```diff
@@ -287,29 +293,49 @@ Then set it as the upstream::
 How do I compare my local repository to a remote repository?
 ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
 
-To display the list of changes that are in your local repository, but not
-in the remote, use::
+You have in your clone a copy of the state of the branches in the remote
+repository.  This state is updated any time you do a ``git pull`` or
+``git fetch``.  You can find out if your local copy is up to date with
+its origin by running::
+
+    $ git remote show origin
+
+If there are commits in any of the remote branches that you haven't
+merged into your local copies of the branches, this command will tell
+you so, and also tell you whether or not you can do a fast-forward merge
+(a ``git pull`` without having to make a separate merge commit).
+
+You can do::
+
+    $ git fetch
+
+to make sure your copies of the remote branches are up to date.
```

Comment:
I'm in the "origin = read-only original, pr = my clone" camp with David - the reason I do it this way is that there are a lot of repos where I first clone them just to tinker with them, and only later do I want to send a PR. At that point, adding a "pr" remote is easier than switching origin to point to my fork instead of upstream. (Being able to do "git checkout master && git pull" to get the latest upstream is then just an added bonus)

The default recommended GitHub workflow makes more sense in a work context (where of course you're going to make changes - that's your job!), but less so when you're exploring open source projects to understand them (and maybe build on top of them).

These are the only two major PR workflows I've encountered though, so I don't think there's a zoo of possibilities - just the default approach, and the common variant of keeping "origin" as upstream.